types/row_proof.go

```diff
@@ -34,6 +34,11 @@ func (rp RowProof) Validate(root []byte) error {
 	if len(rp.Proofs) != len(rp.RowRoots) {
 		return fmt.Errorf("the number of proofs %d must equal the number of row roots %d", len(rp.Proofs), len(rp.RowRoots))
 	}
+	if len(rp.Proofs) != 0 &&
+		(int64(rp.StartRow) != rp.Proofs[0].Index ||
+			int64(rp.EndRow) != rp.Proofs[len(rp.Proofs)-1].Index) {
+		return fmt.Errorf("invalid start/end row")
+	}
```

Comment:
[nit] this could be split into two separate checks with two separate errors: one for start row and one for end row. That way the error message is precise.

Reply:
[6a2148c](https://github.com/celestiaorg/celestia-core/pull/1476/commits/6a2148ccbe65ee40a3a8e507deedb74bbbd322ba)